### PR TITLE
Error ack

### PR DESCRIPTION
**Pull request by chevett**
@mbernardy @zivester ack an error and keep going.  good messages keep processing.  bad ones pile up in a different queue for later processing.

### CODE REVIEW

File: default-exchange-publish.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		ex = _getExchange();
+	}
+
+	return ex.then(function(exchange) {
+		var key = options.key;
+		options = _.extend({}, options);
+		delete options.key;
+
+		exchange.publish(key, msg, options, function(){
+			d.resolve(true);
```
Comment (mbernardy) [inline default-exchange-publish.js]: `d` is undefined here.

File: exchange.js [hunk 2]
```diff
@@ -3,7 +3,9 @@ var q = require('q'),
 	amqp = require('amqp'),
 	Queue = require('./queue'),
 	_ = require('lodash'),
-	EventEmitter = require('events').EventEmitter;
+	EventEmitter = require('events').EventEmitter,
```
Comment (mbernardy) [inline exchange.js]: You don't need to do .EventEmitter
```
[max ~]$ node
> var x = require('events');
undefined
> var y = require('events').EventEmitter;
undefined
> x === y;
true
```

File: default-exchange-publish.js (hunk 1 above)
Comment (chevett) [inline default-exchange-publish.js]: :+1:

File: exchange.js (hunk 2 above)
Comment (chevett) [inline exchange.js]: cool, true for `0.10.x` too?

Comment (zivester) [inline exchange.js]: negative
```
~ $ node --version
v0.10.36
~ $ node
> require('events') === require('events').EventEmitter
false
```

Comment (chevett) [inline exchange.js]: :-1:

**Pull request merged**